src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java

```diff
@@ -127,12 +129,25 @@ private String caCertificate(Map<String, Comparable> properties) {
 
     @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
     private static String readAccountToken() {
-        return readFileContents("/var/run/secrets/kubernetes.io/serviceaccount/token");
+        if (fileExists(ACCOUNT_TOKEN_FILENAME)) {
+            return readFileContents(ACCOUNT_TOKEN_FILENAME);
+        } else {
```

Comment:
I guess we need to warn users if file does not exist and Kubernetes API mode is used.

Reply:
Yeah, or maybe the validation in `validateConfig()`? I think we need to have `token` and `ca-certificate` defined if you use Kubernetes API Mode.

Reply:
I missed the `validateConfig()`. I will add it there